### DISCUSSION TITLE
Reloading aborted scripts

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1411] [Q&A] [answered]
is it normal that a script that was aborted due to missing dependencies causes a total game crash (or fatal game freeze) upon an attempt to reload all scripts?

**Comment by kagikn** [comment]
I don't think that's normal, but how to reproduce? Maybe it's completely responsible to how the script is coded.

**Reply by Michael21107** [reply]
1) download this (1.1.9 currently) https://www.lcpdfr.com/downloads/gta5mods/scripts/39944-simplehud/ 
2) install just the `scripts` folder (there is a nightly and stable option for some reason, I just tested the stable one) and not the props and texture folder OIVs (mentioned dependencies)
3) reload all scripts using `Reload()` at the SHVDN console

**Reply by kagikn** [reply, marked as answer]
Comfirmed in my game. Common unhandled exceptions occured in the `Aborted` event should not crash the game except for fatal ones such as `AccessViolationException`. [The event is supposed to catch common exceptions](https://github.com/scripthookvdotnet/scripthookvdotnet/blob/36a892cb881be67f091d52713854d3224e5c4f1e/source/core/Script.cs#L181-L188), but looks like your report revealed it doesn't properly catch.

Also tell the script author Venoxity to have the script check if the file system watcher (in the `Main` class) is not null before calling `Dispose()`, because although that defect reveals the defect in the aborted event of SHVDN, it is still a defect 🙄

**Reply by Michael21107** [reply]
gotcha, imma tell him, thanks

**Reply by kagikn** [reply]
Since the exception is thrown from a finalizer, the issue is definitely not fault of SHVDN. Exceptions thrown from finalizers can't be caught by try-catch blocks and of course SHVDN can't. I'm gonna have to tell Venoxity how SHVDN can't catch exception thrown from finalizers with technical explanations, or he won't be convinced 😂

I'm sure michael won't be interested, but there's an [article about how exceptions thrown from finalizers will be treated as unhandled exceptions](https://csharp.2000things.com/2012/09/17/672-an-exception-thrown-from-a-finalizer-will-be-treated-as-an-unhandled-exception/) for those interested.

**Reply by Michael21107** [reply]
@kagikn I read the article too, just so you know😂
